### ISSUE TITLE
loadtxt with nested dtype fails with "ValueError: setting an array element with a sequence" (migrated from Trac #1936)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3484]
Original ticket http://projects.scipy.org/numpy/ticket/1936
Reported 2011-08-16 by atmention:gerritholl, assigned to unknown.

I'm sure this used to work. It also works for someone else on IRC.

```
$ python
Python 2.6.6 (r266:84292, Sep 15 2010, 16:22:56) 
[GCC 4.4.5] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from numpy import loadtxt, uint8
>>> from StringIO import StringIO
>>> from numpy.version import version
>>> print version
2.0.0.dev-5cf0a07
>>> loadtxt(StringIO("0 1 2 3"), dtype=[("a", uint8, 2), ("b", uint8, 2)])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/storage4/home/gerrit/.local/lib/python2.6/site-packages/numpy/lib/npyio.py", line 806, in loadtxt
    X = np.array(X, dtype)
ValueError: setting an array element with a sequence.
>>> loadtxt(StringIO("0 1 2 3"), dtype=[("a", uint8, 4)])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/storage4/home/gerrit/.local/lib/python2.6/site-packages/numpy/lib/npyio.py", line 806, in loadtxt
    X = np.array(X, dtype)
ValueError: setting an array element with a sequence.
```

What's going on here?

**Comment by thouis** [comment]
Comment in Trac by atmention:gerritholl, 2011-08-16

**Comment by thouis** [comment]
Comment in Trac by atmention:gerritholl, 2011-08-16

Playing around with 'git bisect' (I'm new here) the culprit appears to be:

```
4e52c48bd37abdce856033f48d3ce9c0a8c5483f is the first bad commit
commit 4e52c48bd37abdce856033f48d3ce9c0a8c5483f
Author: Mark Wiebe <mwwiebeatmention:gmail.com>
Date:   Sun Jun 19 17:41:40 2011 -0500

    BUG: core: Fix corner case of object array assignment (#1870)

    This general aspect of object arrays is fraught with errors, for
    example, eval(repr(x)) produces a 2 instead of 1 dimensional array
    when given the x in the first example of the bug. This fix at least
    makes the result requested in the bug report work.

:040000 040000 72c87734eb0f541c0b0d34663702f4bf55d06915 64fc0029020779a3e535684794c3d2d89cff5868 M  numpy
bisect run success
```

**Comment by thouis** [comment]
Comment in Trac by atmention:gerritholl, 2011-08-16

Replying to [comment:2 gerrit]:

> Playing around with 'git bisect' (I'm new here) the culprit appears to be:

Sorry, I'm probably mistaken, as this is the most recent commit. I'm still new with "git bisect"...

**Comment by thouis** [comment]
Comment in Trac by atmention:gerritholl, 2011-08-16

Replying to [comment:3 gerrit]:

> Replying to [comment:2 gerrit]:
> 
> > Playing around with 'git bisect' (I'm new here) the culprit appears to be:
> 
> Sorry, I'm probably mistaken, as this is the most recent commit. I'm still new with "git bisect"...

Scrap that, I was mistaken. It seems like it /is/ 4e52c48bd37abdce856033f48d3ce9c0a8c5483f sorry. It's obviously not the most recent commit, it only was when I was on the 'git bisect' branch...

**Comment by thouis** [comment]
Comment in Trac by atmention:mwiebe, 2011-08-19

I've fixed the bug, here's the pull request:

https://github.com/numpy/numpy/pull/140

Looks like the bug I fixed in the commit you found was hiding this bug before.

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2011-08-20

fixed in a54a0bdc89dd82ca3cfe.

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2012-04-30
